lib/chef/resource/chocolatey_installer.rb

```diff
@@ -0,0 +1,129 @@
+
+      property :use_native_unzip, [TrueClass, FalseClass], default: false,
+        description: "If set, uses built-in Windows decompression tools instead of 7zip when unpacking the downloaded nupkg. This will be set by default if use_native_unzip is set to a value other than 'false' or '0'. This parameter will be ignored in PS 5+ in favour of using the Expand-Archive built in PowerShell cmdlet directly."
+
+      property :ignore_proxy, [TrueClass, FalseClass], default: false,
+        description: "If set, ignores any configured proxy. This will override any proxy environment variables or parameters. This will be set by default if ignore_proxy is set to a value other than 'false' or '0'."
+
+      property :proxy_url, String,
+        description: "Specifies the proxy URL to use during the download."
+
+      property :proxy_user, String,
+        description: "The username to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_password are set"
+
+      property :proxy_password, String,
+        description: "The password to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_user are set"
+
+      load_current_value do
+        current_state = fetch_choco_installer
+        current_value_does_not_exist! if current_state == false
+        current_state
+      end
+
+      def fetch_choco_installer
+        ::File.exist?("#{ENV["ALLUSERSPROFILE"]}\\chocolatey\\bin\\choco.exe")
+      end
+
+      action :install, description: "Installs Chocolatey package manager" do
+        unless new_resource.download_url.nil?
+          "Set-Item -path env:chocolateyDownloadUrl -Value #{new_resource.download_url}"
+        end
+
+        unless new_resource.chocolatey_version.nil?
+          "Set-Item -path env:chocolateyVersion -Value #{new_resource.chocolatey_version}"
+        end
+
+        if new_resource.use_native_unzip == true
+          "Set-Item -path env:chocolateyUseWindowsCompression -Value true"
+        end
+
+        if new_resource.ignore_proxy == true
+          "Set-Item -path env:chocolateyIgnoreProxy -Value true"
+        end
+
+        unless new_resource.proxy_url.nil?
+          "Set-Item -path env:chocolateyProxyLocation -Value #{new_resource.proxy_url}"
+        end
+
+        if !new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?
+          "Set-Item -path env:chocolateyProxyUser -Value #{new_resource.proxy_user}; Set-Item -path env:chocolateyProxyPassword -Value #{new_resource.proxy_password}"
+        elsif (!new_resource.proxy_user.nil? && new_resource.proxy_password.nil?) || (new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?)
+          Chef::Log.error("Both a Proxy User and a Proxy Password must be set or neither can be set")
```

Comment:
Should this be an exception? More to the point, should this be in `define_resource_requirements`?

Reply:
I have `define_resource_requirements` added